### PR TITLE
Update guzzle6 adapater with httplug new contract

### PR DESCRIPTION
**Pull request by joelwurtz**
I update the guzzle6 adapter to work with the new contract of Httplug, all tests are green.

For the sendRequests method after viewing the guzzle implementation and discussing it with @sagikazarmark on Slack, we concluded there was no real benefits to wrap the Guzzle object as it does not store requests, responses or exceptions objects.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -15,14 +15,22 @@
...
-        "php": ">=5.4",
+        "php": ">=5.5.0",
         "php-http/httplug": "^1.0@dev",
+        "php-http/utils": "^0.1@dev",
         "guzzlehttp/guzzle": "^6.0"
     },
     "require-dev": {
         "ext-curl": "*",
-        "php-http/adapter-integration-tests": "^0.1"
+        "php-http/message-factory": "dev-master as 0.1.1",
```
Comment (sagikazarmark) [inline composer.json]: Why is this necessary? WHere is it used?

File: src/Guzzle6HttpAdapter.php [hunk 2]
```diff
@@ -29,22 +34,27 @@ class Guzzle6HttpAdapter implements HttpAdapter
...
      * {@inheritdoc}
      */
-    public function sendRequest(RequestInterface $request, array $options = [])
+    public function sendRequest(RequestInterface $request)
     {
-        $options = $this->buildOptions($options);
-
         try {
-            return $this->client->send($request, $options);
+            return $this->client->send($request, $this->options);
```
Comment (sagikazarmark) [inline src/Guzzle6HttpAdapter.php]: These options can also be configure in Guzzle at construction time, can't they? If so, we should just wrap the client and forget about options.

File: src/Guzzle6HttpAdapter.php [hunk 3]
```diff
@@ -53,56 +63,42 @@ public function sendRequest(RequestInterface $request, array $options = [])
...
     }
 
     /**
-     * Converts a Guzzle exception into an HttpAdapter exception
+     * Converts a Guzzle exception into an Httplug exception
      *
      * @param RequestException $exception
      *
-     * @return Exception\HttpAdapterException
+     * @return HttpException|NetworkException Return an HttpException if response is available, NetworkException otherwise
```
Comment (sagikazarmark) [inline src/Guzzle6HttpAdapter.php]: In theory, this can be Http\Client\Exception. If we ever change the Exceptions, the method signature remains correct. Doesn't really matter though, since the method is private.

File: composer.json (hunk 1 above)
Comment (joelwurtz) [inline composer.json]: Was used by the test which need discovery service, which need message factory

File: src/Guzzle6HttpAdapter.php (hunk 2 above)
Comment (joelwurtz) [inline src/Guzzle6HttpAdapter.php]: Yes it's the same at the one in the construct of Guzzle, let's remove them.

**Comment by sagikazarmark** [comment]
Thanks @joelwurtz
